src/epoch.rs

```diff
@@ -263,22 +263,77 @@ impl Epoch {
         None
     }
 
-    /// Makes a copy of self and sets the duration and time scale appropriately given the new duration
-    #[must_use]
-    pub fn from_duration(new_duration: Duration, time_scale: TimeScale) -> Self {
-        match time_scale {
-            TimeScale::TAI => Self::from_tai_duration(new_duration),
-            TimeScale::TT => Self::from_tt_duration(new_duration),
-            TimeScale::ET => Self::from_et_duration(new_duration),
-            TimeScale::TDB => Self::from_tdb_duration(new_duration),
-            TimeScale::UTC => Self::from_utc_duration(new_duration),
-            TimeScale::GPST => Self::from_gpst_duration(new_duration),
-            TimeScale::QZSST => Self::from_qzsst_duration(new_duration),
-            TimeScale::GST => Self::from_gst_duration(new_duration),
-            TimeScale::BDT => Self::from_bdt_duration(new_duration),
+    /// Creates an epoch from given duration expressed in given timescale.
+    /// In case of ET, TDB Timescales, a duration since J2000 is expected.
+    #[must_use]
+    pub fn from_duration(new_duration: Duration, ts: TimeScale) -> Self {
```

Comment:
If #237 architecture is adopted, this can simply become:

```rust
pub const fn from_duration(duration: Duration, ts: TimeScale) -> Self {
    Self {
        duration,
        time_scale: ts
    }
```

Then the conversion to/from different time scales only happens when another time scale is requested. For example, if someone did this:

```rust
let e = (366.days() + 1.hours()).ephemeris_time() // Create an ephemeris time epoch that's 2001-01-01T01:00:00 ET (because the ET reference is 01 Jan 2000)
let e_utc = e.into_time_scale(TimeScale::UTC) // This is when the computation between time scales happen
```

Note that I _think_ `into_...` works here given the [Rust naming convention](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv).

As I type this, two other things come to mind:
1. I think it would make sense for `Week` to be a `Unit` so one could do `5.weeks() + 6.hours()`: a week is always exactly 7 days, and I think we didn't add `Week` to `Unit` because it would have been a breaking change
2. The reference epoch should be defined for all TimeScales. The issue still comes with modifiers since there may be some use cases of `365.days().mjd_ephemeris_time()` where the reference epoch is Modified Julian Days... but that adds complexity and rigidity because people can't easily define new reference epochs... Maybe something `365.days().ephemeris_time_from(REF_EPOCH)` where `ref_epoch` is also an epoch that's added to the ephemeris_time reference epoch ... I don't know what's best!